pkg/remote/aws/ec2_eip_association_supplier.go

```diff
@@ -21,11 +22,11 @@ type EC2EipAssociationSupplier struct {
 	runner       *terraform.ParallelResourceReader
 }
 
-func NewEC2EipAssociationSupplier(provider *AWSTerraformProvider) *EC2EipAssociationSupplier {
+func NewEC2EipAssociationSupplier(provider *AWSTerraformProvider, c cache.Cache) *EC2EipAssociationSupplier {
 	return &EC2EipAssociationSupplier{
 		provider,
 		awsdeserializer.NewEC2EipAssociationDeserializer(),
-		repository.NewEC2Repository(provider.session),
+		repository.NewEC2Repository(provider.session, c),
```

Comment:
I wonder if we should build the repository before passing it to supplier instead of passing supplier dependencies ?

Reply:
that way we could also pass the same repository to every client instead of building one each time ?

Reply:
You mean the same way we already do with the S3 repository ? https://github.com/cloudskiff/driftctl/blob/main/pkg/remote/aws/init.go#L29

Reply:
yep exactly